homeassistant/components/nzbget/config_flow.py

```diff
@@ -0,0 +1,143 @@
+"""Config flow for NZBGet."""
+import logging
+from typing import Any, Dict, Optional
+
+import voluptuous as vol
+
+from homeassistant.config_entries import CONN_CLASS_LOCAL_POLL, ConfigFlow, OptionsFlow
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_NAME,
+    CONF_PASSWORD,
+    CONF_PORT,
+    CONF_SCAN_INTERVAL,
+    CONF_SSL,
+    CONF_USERNAME,
+    CONF_VERIFY_SSL,
+)
+from homeassistant.core import callback
+from homeassistant.helpers.typing import ConfigType, HomeAssistantType
+
+from .const import (
+    DEFAULT_NAME,
+    DEFAULT_PORT,
+    DEFAULT_SCAN_INTERVAL,
+    DEFAULT_SSL,
+    DEFAULT_VERIFY_SSL,
+)
+from .const import DOMAIN  # pylint: disable=unused-import
+from .coordinator import NZBGetAPI, NZBGetAPIException
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def validate_input(hass: HomeAssistantType, data: dict) -> Dict[str, Any]:
+    """Validate the user input allows us to connect.
+
+    Data has the keys from DATA_SCHEMA with values provided by the user.
+    """
+    nzbget_api = NZBGetAPI(
+        data[CONF_HOST],
+        data[CONF_USERNAME] if data[CONF_USERNAME] != "" else None,
+        data[CONF_PASSWORD] if data[CONF_PASSWORD] != "" else None,
+        data[CONF_SSL],
+        data[CONF_VERIFY_SSL],
+        data[CONF_PORT],
+    )
+
+    nzbget_api.version()
+
+    return True
+
+
+class NZBGetConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for NZBGet."""
+
+    VERSION = 1
+    CONNECTION_CLASS = CONN_CLASS_LOCAL_POLL
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry):
+        """Get the options flow for this handler."""
+        return NZBGetOptionsFlowHandler(config_entry)
+
+    async def async_step_import(
+        self, user_input: Optional[ConfigType] = None
+    ) -> Dict[str, Any]:
+        """Handle a flow initiated by configuration file."""
+        if CONF_SCAN_INTERVAL in user_input:
+            user_input[CONF_SCAN_INTERVAL] = user_input[CONF_SCAN_INTERVAL].seconds
+
+        return await self.async_step_user(user_input)
+
+    async def async_step_user(
+        self, user_input: Optional[ConfigType] = None
+    ) -> Dict[str, Any]:
+        """Handle a flow initiated by the user."""
+        if self._async_current_entries():
+            return self.async_abort(reason="single_instance_allowed")
+
+        if user_input is None:
```

Comment:
It's better to invert this check and end the step with the user form.

Reply:
WIP here https://github.com/home-assistant/core/pull/39425